articles/introduction-to-android-views-and-view-groups/index.md

```diff
@@ -0,0 +1,352 @@
+- `android:textColor=”#de4f3e” ` - it is used to change the color of the text.
+- `android:textColorHint=”#de4f3e” ` - it is used to change the text color of hint text.
+- `android:inputType=”text” `- it is used to specify the type as "text" which is placed in the text fields.
+- `android:maxLength=”10”` - specifies how many characters can be input. in this case, it only accepts 10 characters.
+
+**Button**
+it is a user interface feature that contains a text or an icon that the user can tap or click to act.
+
+Example of a Button view XML code snippet:
+
+```xml  
+    <Button
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="click me"
+        android:backgroundTint="#df3e4c"/>
+```
+
+![Output screen](/engineering-education/android-views-and-viewgroups/screenshot-button.png)
+
+The attributes are defined as follows:
+
+- `android:backgroundTint=”#de4f3e”` - sets the color of the text in the button.
+- `android:text=”click me”` - is the text that will be displayed by the button
+
+**ImageView**
+it is a widget used to display images and bitmaps in android applications. Bitmap files supported by android are in three formats:
+
+`.png` -preferred
+`.jpg` -acceptable
+` .gif` -discouraged
+
+Example of ImageView XML code snippet:
+
+```xml
+    <ImageView
+        android:layout_width="160dp"
+        android:layout_height="140dp"
+        android:src="@drawable/ic_campaign"/>
+```
+![Output screen](/engineering-education/android-views-and-viewgroups/screenshot-image-view.png)
+
+The attributes are defined as follows:
+
+`android:src=”@drawable/ic_campaign” />` - sets a drawable as the content of this ImageView.
+A drawable resource is a general term for a graph that can be drawn on the screen and can be retrieved by adding attributes such as `android:drawable` and `android:icon` to another XML resource.
+
+Android also provides more views such as Spinners, CheckBox and RadioButtons for more user input
+Output screen
+
+#### ViewGroups
+A ViewGroup is a view that can hold more views ( child class). The View group is the base class for containers, templates and views. Views use `LayoutParams` that serves to inform the parent class how to lay them out. For an easier understanding of viewgroups, we'll use the views we discussed to illustrate how we can align each of them in the same viewgroup. Some of the layout parameters take a boolean value, such as "true" or "false".
+
+Some of the commonly used Viewgroups are:
+- RelativeLayout
+- LinearLayout
+- ConstraintLayout
+- FrameLayout
+
+**RelativeLayout**
+RelativeLayout displays views are relative to the others. The view position has some relation to the sibling.  You might be able to substitute many nested LinearLayout groups with a single RelativeLayout if you find yourself using them because it can delete nested view groups and keep the hierarchy of your layout flat, which improves performance.
```

Comment:
"RelativeLayout displays views relatively to one another" flows better. Rephrase the second sentence. It is too long.